### PR TITLE
Point-Supervised Loss Function

### PR DESCRIPTION
**Pull request by poulguoskov**
This branch is created from the `3-simulate-clicks---weak-label` branch.

Added PartialCrossEntropyLoss for point supervision and simple test.

### CODE REVIEW

**Comment by poulguoskov** [comment]
## Next
- add more complex losses
- update dataloader
- new training script

**Comment by poulguoskov** [comment]
# @kajahovinbole
We can add a small helper to the simulator to output the exact tensors the point-loss needs. Something like:
```python
def to_loss_tensors(self, mask, pos_points, neg_points):
    pos_mask, neg_mask = self.to_click_masks(mask, pos_points, neg_points)
    click_mask = (pos_mask + neg_mask).clamp(max=1)
    click_labels = pos_mask
    return click_mask, click_labels
```
Then the training code can directly do:
```python
pos_pts, neg_pts = sim.sample_clicks(mask)
click_mask, click_labels = sim.to_loss_tensors(mask, pos_pts, neg_pts)
loss = click_loss(pred, click_mask, click_labels)
```
Would you be okay with adding this to the simulator?

**Review by kajahovinbole — APPROVED** [review]
